tests/system/utils.py

```diff
@@ -39,3 +40,121 @@ def get_resource(resource):
                 raise Exception
         except Exception:
             raise DCOSException("Can't read from resource: {0}. Please check that it exists.".format(resource))
+
+
+class FaultDomain:
+    """
+    High-level representation of a fault domain
+    """
+    def __init__(self, config):
+        # Make sure config is a dict
+        if not type(config) is dict:
+            config = {}
+
+        # Extract fault domain
+        fault_domain = config.get('fault_domain', {})
+
+        # Extract zone/region
+        self.zone = fault_domain.get('zone', {}).get('name', 'default')
+        self.region = fault_domain.get('region', {}).get('name', 'default')
+
+
+def get_cluster_local_domain():
+    """Contacts the DC/OS mesos master and returns it's faultDomain configuration (aka "local domain")
+    """
+    master = mesos.get_master()
+    return FaultDomain(master.state().get('domain', {}))
+
+
+def get_cluster_agent_domains():
+    """Returns a dictionary with the slave IDs in the cluster and their corresponding
+       fault domain information
+    """
+    slave_domains = {}
```

Comment:
Please use `agent_domains`.

Reply:
_Not my fault that the API uses `master.slaves` as name 🤣_
Done